apps/oxlint/src/lint.rs

```diff
@@ -280,6 +281,32 @@ impl Runner for LintRunner {
 
         let lint_config = config_builder.build();
 
+        // Enable type-aware linting if:
+        // 1) `--tsconfig` is passed
+        // 2) `tsgolint` appears to be installed
+        let type_aware = basic_options.tsconfig.is_some()
+            && (
+                // Check if `tsgolint` is executable
+                std::process::Command::new("tsgolint").arg("--help").output().is_ok()
```

Comment:
can we add a TODO comment here, if `tsconfig` is supplied, and type aware rules are enabled, and no tsgolint binary is found, we should return an error else it'll be confusing to users